### ISSUE TITLE
Document recent updates to the templating process

### ISSUE THREAD

**Issue by choldgraf** [jupyter/nbconvert#1119]
In #1056 there were some nice updates to how users can define templates and extend nbconvert. I looked through and didn't see any changes to the documentation to explain this. We should document these changes! Is there a plan for how to guide users through this new process?

**Comment by MSeal** [comment]
I agree 100%. Any volunteers to get the first pass written @SylvainCorlay @maartenbreddels @t-makaro @choldgraf ?

**Comment by choldgraf** [comment]
I'm happy to write up some docs although I must admit I'm not positive what are the things that have changed :-)  is it safe to say that we could just port over the voila templating docs directly, more or less? If not, maybe @maartenbreddels can give a brief synopsis of the pieces that should be changed and I can try to flesh them out?
